File: csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Linq;

public static class RnaTranscription
{
    public static string ToRna(string nucleotide) =>
        new(nucleotide.Select(c => c switch
        {
            'G' => 'C',
            'C' => 'G',
            'T' => 'A',
            'A' => 'U',
            _ => throw new ArgumentException()
        }).ToArray());
}

File: vbnet/rna-transcription/RnaTranscription.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Public Module RnaTranscription
    Private DnaToRna As Dictionary(Of Char, Char) = New Dictionary(Of Char, Char) From {
        {"A"c, "U"c},
        {"C"c, "G"c},
        {"G"c, "C"c},
        {"T"c, "A"c}
    }
    Public Function ToRna(ByVal nucleotide As String) As String
        Return String.Concat(nucleotide.Select(Function(c) DnaToRna(c)))
    End Function
End Module
